src/ui/public/vislib/lib/x_axis.js

```diff
@@ -351,7 +351,11 @@ export default function XAxisFactory(Private) {
         .attr('dy', '-.60em')
         .attr('transform', function rotate() {
           return 'rotate(-90)';
-        });
+        })
+          .append('title')
+          .text(function () {
+            return this.__data__;
```

Comment:
My d3 is a little rusty, but I'm not sure __data__ is the most appropriate way to access the text here. Won't the first parameter passed to this function contain the text you need?

Also, indentation needs fixed.